core/services/ardupilot_manager/mavlink_proxy/test_all.py

```diff
@@ -144,6 +145,11 @@ def run_common_routing_tests(
         assert router.start(endpoint), f"Failed to start {router.name()} with endpoint {endpoint}."
         assert router.is_running(), f"{router.name()} is not running after start."
         assert router.exit(), f"{router.name()} could not stop."
+        for _ in range(5):
+            if not router.is_running():
+                break
+            time.sleep(1)
+        assert not router.is_running(), f"{router.name()} is not stopping after exit."
```

Comment:
I'm using a logic here to give it some chances, but praticaly speaking, it usually takes some milliseconds to finish, so one `time.sleep(1)` instead of the for loop is probably enough for 99% of the time.

Let me know what you prefer.